pandas/core/generic.py

```diff
@@ -5291,47 +5288,162 @@ def replace(self, to_replace=None, value=None, inplace=False, limit=None,
                'piecewise_polynomial' in scipy 0.18; backwards-compatible with
                scipy < 0.18
 
-        axis : {0, 1}, default 0
-            * 0: fill column-by-column
-            * 1: fill row-by-row
-        limit : int, default None.
-            Maximum number of consecutive NaNs to fill. Must be greater than 0.
+        axis : {0 or 'index', 1 or 'columns', None}, default None
+            Axis to interpolate along.
+        limit : int, optional
+            Maximum number of consecutive NaNs to fill. Must be greater than
+            0.
+        inplace : bool, default False
+            Update the data in place if possible.
         limit_direction : {'forward', 'backward', 'both'}, default 'forward'
-        limit_area : {'inside', 'outside'}, default None
-            * None: (default) no fill restriction
-            * 'inside' Only fill NaNs surrounded by valid values (interpolate).
-            * 'outside' Only fill NaNs outside valid values (extrapolate).
-            .. versionadded:: 0.21.0
-
             If limit is specified, consecutive NaNs will be filled in this
             direction.
-        inplace : bool, default False
-            Update the NDFrame in place if possible.
+        limit_area : {`None`, 'inside', 'outside'}
+            If limit is specified, consecutive NaNs will be filled with this
+            restriction.
+
+            * None: No fill restriction (default).
+            * 'inside': Only fill NaNs surrounded by valid values
+              (interpolate).
+            * 'outside': Only fill NaNs outside valid values (extrapolate).
+
+            .. versionadded:: 0.21.0
+
         downcast : optional, 'infer' or None, defaults to None
             Downcast dtypes if possible.
-        kwargs : keyword arguments to pass on to the interpolating function.
+        **kwargs
+            Keyword arguments to pass on to the interpolating function.
 
         Returns
         -------
-        Series or DataFrame of same shape interpolated at the NaNs
+        Series or DataFrame
+            Same-shape object interpolated at the NaN values
 
         See Also
         --------
-        reindex, replace, fillna
+        replace : replace a value
+        fillna : fill missing values
+        scipy.interpolate.Akima1DInterpolator : piecewise cubic polynomials
+            (Akima interpolator)
+        scipy.interpolate.BPoly.from_derivatives : piecewise polynomial in the
+            Bernstein basis
+        scipy.interpolate.interp1d : interpolate a 1-D function
+        scipy.interpolate.KroghInterpolator : interpolate polynomial (Krogh
+            interpolator)
+        scipy.interpolate.PchipInterpolator : PCHIP 1-d monotonic cubic
+            interpolation
+        scipy.interpolate.CubicSpline : cubic spline data interpolator
+
+        Notes
+        -----
+        If the selected `method` is one of 'krogh', 'piecewise_polynomial',
```

Comment:
"The 'krogh', 'piecewise_polynomial', ... methods are wrappers around the respective SciPy implementations" would be better wording